### PR TITLE
1619 Change List Presence Tag indexing

### PR DESCRIPTION
**Pull request by mwfrost**
Closes: #1619 

This adds a custom analyzer to the ListPresenceTag indexing code - now each list presence tag name with an underscore is split into multiple tokens, and the search will match any one of the tokens. 

After checking out this branch, use the admin interface to create a new ListPresenceTag with "animal_feed" as its name and slug, and run `python manage.py search_index --rebuild` to rebuild the index. Then search for "animal" and confirm that both the "animal" and "animal_feed" tags appear on the "Tag" results page.

### CODE REVIEW

File: dashboard/fixtures/14_list_presence_tag.yaml [hunk 1]
```diff
@@ -2997,3 +2997,11 @@
     name: writing
     slug: writing
     definition: Voluptatem eius numquam sit consectetur neque.
+- model: dashboard.extractedlistpresencetag
+  pk: 374
```
Comment (mwfrost) [inline dashboard/fixtures/14_list_presence_tag.yaml:3001]: This was just added for a test

File: dashboard/tests/functional/test_search.py [hunk 2]
```diff
@@ -375,6 +375,16 @@ def test_search_tag(self):
         total_took = response_html.xpath('normalize-space(//*[@id="total-took"])')
         self.assertIn(expected_total, total_took)
 
+        # underscored tag names should be tokenized
+        name = "animal"
+        qs = self._get_query_str(name)
+        response = self.client.get("/search/tag/" + qs)
+        soup = bs4.BeautifulSoup(response.content, features="lxml")
+        found = soup.find_all("small", {"id": "total-took"})
+        self.assertInHTML('<small class="text-muted pb-4" id="total-took">2 tags returned in 0.001 seconds.</small>', str(found[0]))
```
Comment (hausman-gdit) [inline dashboard/tests/functional/test_search.py:384]: specifying seconds to return fails the test (passes when `0.0`), looking for alternatives/checking more into pytest.

**Review by hausman-gdit — CHANGES_REQUESTED** [review]
Test Fix: specifying seconds to return fails the test (passes when `0.0` for me not `0.001`)

File: dashboard/tests/functional/test_search.py (hunk 2 above)
Comment (mwfrost) [reply dashboard/tests/functional/test_search.py]: Try `manage.py test dashboard.tests.functional.test_search.TestSearch.test_search_tag` now

File: dashboard/tests/functional/test_search.py [hunk 3]
```diff
@@ -375,6 +375,20 @@ def test_search_tag(self):
...
         self.assertIn(expected_total, total_took)
 
+        # underscored tag names should be tokenized
+        name = "animal"
+        qs = self._get_query_str(name)
+        response = self.client.get("/search/tag/" + qs)
+        soup = bs4.BeautifulSoup(response.content, features="lxml")
+        found = soup.find_all("small", {"id": "total-took"})
+        # The actual time might vary across tests, so alter the string to exclude the digits
+        found_string = re.sub("\d+.\d+ seconds", "x.x seconds", str(found[0]))
```
Comment (hausman-gdit) [inline dashboard/tests/functional/test_search.py:385]: lol clever!

**Review by hausman-gdit — APPROVED** [review]
PR resolves issue, tests pass post-fix

**Pull request merged**